### PR TITLE
MikuMikuDance Format

### PR DESCRIPTION
**Pull request by aoowweenn**
From #644, try to import .pmx, .vmd, and .pmd file.

Bug fixed
- Exporter flag should be same with the importer's in the [qt viewer](https://github.com/assimp/assimp/pull/1212/commits/ba357e74d274e7847e31a41e5aba2a976d81d9fd)
- [Assbin loader](https://github.com/assimp/assimp/pull/1212/commits/845d206959993392a08ab756a78246cbae96c9c0) should handle mParent fields in node hierarchy.
  - I faced the problem to import .pmx file to Urho3d since they manually revised the assimp repo, so first convert .pmx to .assbin and import into Urho3d.

Status
- .pmx file: mesh, texture, skeleton
- Create local DevIL cmake module to avoid finding ILUT

TODO
- [#961, #1000] patch COLLADA exportor to verify bone weights
- import animation from .vmd file

Pending
- physics: spring bones, IK, soft bodies
- .pmd file

### CODE REVIEW

**Comment by kimkulling** [comment]
Build is broken, could you fix it?

File: code/MMDVmdParser.h [hunk 1]
```diff
@@ -0,0 +1,367 @@
...
+
+	/// IKの有効無効
+	class VmdIkEnable
+	{
+	public:
+		std::string ik_name;
+		bool enable;
+	};
+
+	/// IKフレーム
```
Comment (kimkulling) [inline code/MMDVmdParser.h]: False encoding

**Comment by kimkulling** [comment]
I looked over your pull request: good so far. But there are unreadable comments with a false encoding. Please only englich comments! Thanks!

**Comment by aoowweenn** [comment]
The parser is almost not my work though, I may find some time to revise those Japanese comments...

By the way, morphing is possibly the next one that can be added to the PMX importer,
There are still some problems with the VMD importer since no IK, no full skeleton hierarchy, and position offset against PMD/PMX bind pose.
I've tried to merge a PMX bind pose and a VMD file to .ani format used in Urho3D.
The result is not good but it seems that I got it basically right except:

1. Duplicate bones merging
2. No 2 feet IK
3. Right frame interpolation

Besides, I think the DAE importer is strange,
UTF-8 is XML natively supported, but why filtering out non-ASCII characters?

**Comment by acgessler** [comment]
What's the license status of MMDParser? It doesn't tell and you stated it's not your work. Is it BSD-3 clause compliant?

**Comment by aoowweenn** [comment]
It's under CC0 1.0 so I think there should be no problem with BSD.
Also the author says,

> You can use this library freely. I dont't care about your usages.

[link](https://github.com/oguna/MMDFormats)

I had contributed non-Windows encoding part to his repo,
and in Assimp I used the internal unicode library instead for both Win/*nix.

**Comment by kimkulling** [comment]
Sorry, I completly forgot about this pull-request. When all license info is there we can merge this guy I guess. Could you just merge the latest master into it? Thanks a lot for all the great work and sorry for my carelessness!

**Comment by kimkulling** [comment]
Oh, we can merge this.

**Comment by aoowweenn** [comment]
Thanks,
hope this contribution is useful for others.

**Pull request merged**